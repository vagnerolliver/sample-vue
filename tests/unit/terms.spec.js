import Terms from '@/pages/Terms.vue'
import { mount } from "@vue/test-utils";

describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Terms)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('snapshot renders correctly', () => {
    const wrapper = mount(Terms)
    expect(wrapper.element).toMatchSnapshot()
  })
})
