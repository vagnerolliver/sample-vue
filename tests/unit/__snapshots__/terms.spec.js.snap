// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot renders correctly 1`] = `
<section>
  <div
    class="field"
  >
    <label
      class="label"
    >
      Name
    </label>
     
    <div
      class="control"
    >
      <input
        class="input"
        placeholder="Text input"
        type="text"
      />
    </div>
  </div>
   
  <div
    class="field"
  >
    <label
      class="label"
    >
      Username
    </label>
     
    <div
      class="control has-icons-left has-icons-right"
    >
      <input
        class="input is-success"
        placeholder="Text input"
        type="text"
        value="bulma"
      />
       
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-user"
        />
      </span>
       
      <span
        class="icon is-small is-right"
      >
        <i
          class="fas fa-check"
        />
      </span>
    </div>
     
    <p
      class="help is-success"
    >
      This username is available
    </p>
  </div>
   
  <div
    class="field"
  >
    <label
      class="label"
    >
      Email
    </label>
     
    <div
      class="control has-icons-left has-icons-right"
    >
      <input
        class="input is-danger"
        placeholder="Email input"
        type="email"
        value="hello@"
      />
       
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-envelope"
        />
      </span>
       
      <span
        class="icon is-small is-right"
      >
        <i
          class="fas fa-exclamation-triangle"
        />
      </span>
    </div>
     
    <p
      class="help is-danger"
    >
      This email is invalid
    </p>
  </div>
   
  <div
    class="field"
  >
    <label
      class="label"
    >
      Subject
    </label>
     
    <div
      class="control"
    >
      <div
        class="select"
      >
        <select>
          <option>
            Select dropdown
          </option>
           
          <option>
            With options
          </option>
        </select>
      </div>
    </div>
  </div>
   
  <div
    class="field"
  >
    <label
      class="label"
    >
      Message
    </label>
     
    <div
      class="control"
    >
      <textarea
        class="textarea"
        placeholder="Textarea"
      />
    </div>
  </div>
   
  <div
    class="field"
  >
    <div
      class="control"
    >
      <label
        class="checkbox"
      >
        <input
          type="checkbox"
        />
        
        I agree to the 
        <a
          href="#"
        >
          terms and conditions
        </a>
      </label>
    </div>
  </div>
   
  <div
    class="field"
  >
    <div
      class="control"
    >
      <label
        class="radio"
      >
        <input
          name="question"
          type="radio"
        />
        
        Yes
      
      </label>
       
      <label
        class="radio"
      >
        <input
          name="question"
          type="radio"
        />
        
        No
      
      </label>
    </div>
  </div>
   
  <div
    class="field is-grouped"
  >
    <div
      class="control"
    >
      <button
        class="button is-link"
      >
        Submit
      </button>
    </div>
     
    <div
      class="control"
    >
      <button
        class="button is-text"
      >
        Cancel
      </button>
    </div>
  </div>
</section>
`;
